streams/src/main/java/org/apache/kafka/streams/kstream/ApiUtils.java

```
@@ -0,0 +1,33 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.kstream;
+
+import java.time.Duration;
+
+final class ApiUtils {
+    private ApiUtils() {}
+
+    static Duration validateMillisecondDuration(final Duration duration, final String message) {
```

It's possible to pass in a Duration that's bigger than max long ms. (For example, `Duration.ofSeconds(Long.MAX_VALUE)`). If this happens, Duration will throw an `ArithmeticException` when we try to get millis via `Duration#toMillis()`.

I would like to fail fast by checking the bounds on the `Duration` as close to the user code as possible, so I've created this utility method for use in the DSL and other user-facing APIs.